docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,274 @@
+
+BigDL Orca example code can be found here [Fashion_MNIST Example](https://github.com/intel-analytics/BigDL/tree/main/docs/docs/tutorials/tutorial_example/Fashion_MNIST/).
+
+# Key Concepts
+## OrcaContext
+A BigDL Orca program usually starts with the initialization of OrcaContext. We can specify the `runtime` (spark or ray) and `cluster_mode` arguments to create or get a SparkContext or RayContext with optimized configurations for BigDL performance.
+
+## Orca Estimator
+After initializing the OrcaContext, we could simply create an Estimatornow, and the Estimator will replicate the model on each node in the cluster, feed the data partition on each node to the local model replica, and synchronize the model parameters using various backend technologies.
+
+Let's get started!
+# Prepare Environment
+Before submitting the BigDL program to Yarn Cluster, we need to setup the environment as the following steps:
+* Setup JAVA Environment
+* Setup Spark Environment
+* Setup Hadoop Environment
+* Install All Needed Python Libraries 
+* For CDH Users
+
+## Setup JAVA Environment
+We need to download and install JDK in the environment, and properly set the environment variable JAVA_HOME, which is required by Spark. JDK8 is highly recommended.
+
+```bash
+# For Ubuntu
+sudo apt-get install openjdk-8-jre
+export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64/
+
+# For CentOS
+su -c "yum install java-1.8.0-openjdk"
+export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.282.b08-1.el7_9.x86_64/jre
+
+export PATH=$PATH:$JAVA_HOME/bin
+java -version  # Verify the version of JDK.
+```
+
+## Setup Spark Environment
+We need to set environment variables `${SPARK_HOME}` as follows:
+```bash
+export SPARK_HOME=the folder path where you extract the Spark package
+```
+
+## Setup Hadoop Environment
+Check the Hadoop setup and configurations of our cluster. Make sure we correctly set the environment variable HADOOP_CONF_DIR, which is needed to initialize Spark on YARN:
+
+```bash
+export HADOOP_CONF_DIR=the directory of the hadoop and yarn configurations
+```
+
+## Install Needed Python Libraries
+We need first to use conda to prepare the Python environment on the local machine where we submit our application. Create a conda environment, install BigDL and all the needed Python libraries in the created conda environment:
+``` bash
+conda create -n bigdl python=3.7  # "bigdl" is conda environment name, you can use any name you like.
+conda activate bigdl
+# Use conda or pip to install all the needed Python dependencies in the created conda environment.
+pip install bigdl
+pip install conda-pack # Use conda-pack to pack the conda environment to an archive
+```
+### Options
+* When you are running a program on Ray backend, please install Ray as below:
+```bash
+pip install ray[default]
+```
+* When you are running a TensorFlow2 model, please install TensorFlow as below:
+```bash
+pip install tensorflow==2.6.0 keras==2.6.0 # When you are running a tensorflow model.
```

Comment:
TnesorFlow, PyTorch, etc., are just extra python dependency that need to be installed, yes?

Reply:
yes, bigdl will automatically install torch. As for tensorflow, user only need to install it when running tf models.

Reply:
Why does bigdl automatically install torch @yangw1234 @hkvision ?

Reply:
I don't think we will automatically install torch?

Reply:
Actually we will.

Reply:
@jason-dai `pip install bigdl-orca` will not install torch. `pip install bigdl` will because `bigdl` depends on `bigdl-chronos` which depends on `bigdl-nano[pytorch]`

Reply:
> @jason-dai `pip install bigdl-orca` will not install torch. `pip install bigdl` will because `bigdl` depends on `bigdl-chronos` which depends on `bigdl-nano[pytorch]`

@TheaperDeng Does it make sense? Should `bigdl-chronos` depend on `bigdl-nano[pytorch]`? Or make `bigdl-chronos[pytorch]` depend on `bigdl-nano[pytorch]`?

Reply:
> > @jason-dai `pip install bigdl-orca` will not install torch. `pip install bigdl` will because `bigdl` depends on `bigdl-chronos` which depends on `bigdl-nano[pytorch]`
> 
> @TheaperDeng Does it make sense? Should `bigdl-chronos` depend on `bigdl-nano[pytorch]`? Or make `bigdl-chronos[pytorch]` depend on `bigdl-nano[pytorch]`?

I can make `pip install bigdl-chronos` does not install `bigdl-nano[pytorch]`. The only concern is that most of the function in bigdl-chronos can not be enabled without `bigdl-nano[pytorch]`.

Reply:
btw, do we really need `bigdl-chronos` in `pip install bigdl`?

Reply:
`pip install bigdl` is supposed to install all components of BigDL; I think the question is how we want to deal with PyTorch and TensorFlow dependency when installing BigDL (or its components).

Reply:
> btw, do we really need `bigdl-chronos` in `pip install bigdl`?

`pip install bigdl` is an easy way to install all the BigDL packages including bigdl-nano, bigdl-dllib, bigdl-orca, bigdl-chronos, bigdl-friesian, bigdl-serving and their dependencies [bigdl install guide](https://bigdl.readthedocs.io/en/latest/doc/UserGuide/python.html#official-release).

For Orca user only, they could use `pip install bigdl-orca`